src/game/common/rts/team.cpp

```diff
@@ -0,0 +1,817 @@
+    max = d->Get_Int(g_teamUnitMaxCount7Key, &exists);
+    str = d->Get_AsciiString(g_teamUnitType7Key, &exists);
+
+    if (max > 0 && exists) {
+        m_unitsInfo[m_numUnitsInfo].min_units = min;
+        m_unitsInfo[m_numUnitsInfo].max_units = max;
+        m_unitsInfo[m_numUnitsInfo].unit_thing_name = str;
+        m_numUnitsInfo++;
+    }
+
+    str = d->Get_AsciiString(g_teamHomeKey, &exists);
+
+    m_homeLocation.x = 0.0f;
+    m_homeLocation.y = 0.0f;
+    m_homeLocation.z = 0.0f;
+    m_hasHomeLocation = false;
+
+    if (exists) {
+        for (const Waypoint *i = g_theTerrainLogic->Get_First_Waypoint(); i != nullptr; i = i->Get_Next()) {
+            if (i->Get_Name() == str) {
+                m_homeLocation = *i->Get_Location();
+                m_hasHomeLocation = true;
+            }
+        }
+    }
+
+    m_scriptOnCreate = d->Get_AsciiString(g_teamOnCreateScriptKey, &exists);
+    m_isAIRecruitable = d->Get_Bool(g_teamIsAIRecruitableKey, &exists);
+
+    if (!exists) {
+        m_isAIRecruitable = false;
+    }
+
+    m_isBaseDefense = d->Get_Bool(g_teamIsBaseDefenseKey, &exists);
+    m_isPerimeterDefense = d->Get_Bool(g_teamIsPerimeterDefenseKey, &exists);
+    m_automaticallyReinforce = d->Get_Bool(g_teamAutoReinforceKey, &exists);
+    int aggressiveness = d->Get_Bool(g_teamAggressivenessKey, &exists);
+    m_aggressiveness = 0;
+
+    if (exists) {
+        m_aggressiveness = aggressiveness;
+    }
+
+    m_transportsReturn = d->Get_Bool(g_teamTransportsReturnKey, &exists);
+    m_avoidThreats = d->Get_Bool(g_teamAvoidThreatsKey, &exists);
+    m_attackCommonTarget = d->Get_Bool(g_teamAttackCommonTargetKey, &exists);
+    m_maxInstances = d->Get_Int(g_teamMaxInstancesKey, &exists);
+    m_scriptOnIdle = d->Get_AsciiString(g_teamOnIdleScriptKey, &exists);
+    m_initialIdleFrames = d->Get_Int(g_teamInitialIdleFramesKey, &exists);
+    m_scriptOnEnemySighted = d->Get_AsciiString(g_teamEnemySightedScriptKey, &exists);
+    m_scriptOnAllClear = d->Get_AsciiString(g_teamAllClearScriptKey, &exists);
+    m_scriptOnDestroyed = d->Get_AsciiString(g_teamOnDestroyedScriptKey, &exists);
+    m_destroyedThreshold = d->Get_Real(g_teamDestroyedThresholdKey, &exists);
+    m_scriptOnUnitDestroyed = d->Get_AsciiString(g_teamOnUnitDestroyedScriptKey, &exists);
+    m_productionPriority = d->Get_Int(g_teamProductionPriorityKey, &exists);
+    m_productionPrioritySuccessIncrease = d->Get_Int(g_teamProductionPrioritySuccessIncreaseKey, &exists);
+    m_productionPriorityFailureDecrease = d->Get_Int(g_teamProductionPriorityFailureDecreaseKey, &exists);
+    m_productionCondition = d->Get_AsciiString(g_teamProductionConditionKey, &exists);
+    m_executesActionsOnCreate = d->Get_Bool(g_teamExecutesActionsOnCreateKey, &exists);
+
+    for (int key = 0; key < ARRAY_SIZE(m_genericScripts); key++) {
+        Utf8String str;
+        str.Format("%s%d", g_theNameKeyGenerator->Key_To_Name(g_teamGenericScriptHookKey).Str(), key);
```

Comment:
Can split this into 2 lines for readability.

Reply:
clang-format won't let me split it.

Reply:
You can write

```c++
Utf8String name = g_theNameKeyGenerator->Key_To_Name(g_teamGenericScriptHookKey);
str.Format("%s%d", name.Str(), key);
```

Much easier to read this way.

Reply:
fixed.